crates/autopilot/src/infra/solvers/mod.rs

```diff
@@ -42,9 +42,31 @@ impl Driver {
         &self,
         request: &settle::Request,
         timeout: std::time::Duration,
-    ) -> Result<settle::Response> {
-        self.request_response("settle", request, Some(timeout))
+    ) -> Result<()> {
+        let url = util::join(&self.url, "settle");
```

Comment:
This is a little bit dangerous because you assume the `self.url` will always have a slash, but it is not enforced anywhere (I didn't see it at least :thinking: ). If a driver config is added without the slash this will crash. Maybe we could check if the last character is a slash to ensure this :thinking:

Reply:
`util::join` actually enforces exactly that because we ran into the issues you mentioned in the past.

Reply:
aaah, I didn't see it! I am blind! thanks for the clarification @MartinquaXD :ok_hand: :pray: